### PR TITLE
feat: Like history pages

### PR DESCRIPTION
**Pull request by rutu14**
### Like Page
- On clicking the heart btn, if user is not logged in, it redirects to login page.
- If logged in the color of btn changes and the video card is visible in liked videos page.
- On clicking the liked btn, the video is removed from liked videos.

### History Page
- If user is logged in and presses the play btn, the video is noted in history page.
- If the user is not logged in, the video is not noted.

[Click here to view the files](https://github.com/rutu14/Video-Library-React/pull/5/commits/d38bf94ab3d1de275d16035350d36f725659653e)

### CODE REVIEW

**Comment by iRohitGaur** [comment]
UI looks really good.

UI suggestion:
- the buttons on the card are changing position based on the screen size. Please look into it.

File: src/context/historyContext.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+const HistoryVideoContext = createContext(defaultValue);
+
+const HistoryVideoProvider = ({ children }) => {
+
+    const [ state, dispatch  ] = useReducer(historyVideoReducer, defaultValue);
+
+    const getHistoryVideo = async () => {
+        try {
+            dispatch({ type: "HISTORY_VIDEO_REQUEST" })
+            const token = localStorage.getItem("token");
```
Comment (iRohitGaur) [inline src/context/historyContext.js:21]: suggestion: call to localStorage is synchronous so it is better to extract this data once and reuse it.

File: src/context/historyContext.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+            const config = { headers: { 'authorization': token } };
+            const { data } = await axios.get( '/api/user/history', config)
+            console.log( data.history )
```
Comment (iRohitGaur) [inline src/context/historyContext.js:24]: suggestion: remove `console.log`

File: src/context/historyContext.js [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+            dispatch({ type: "HISTORY_VIDEO_ERROR", payload: error })
+        }
+    }
+
+    const deleteAllHistoryVideo = async ( videoId ) => {
+        try {
+            dispatch({ type: "HISTORY_VIDEO_REQUEST" })
+            const token = localStorage.getItem("token");
+            const config = { headers: { 'authorization': token } };
+            const { data } = await axios.delete( '/api/user/history/all', config );
```
Comment (iRohitGaur) [inline src/context/historyContext.js:61]: suggestion: do error handling

**Review by iRohitGaur — APPROVED** [review]
Nice work!

Gave some suggestions, please look into it.

Good to merge with changes 👍🏻

**Review by sanu29 — APPROVED** [review]
All good to proceed with commit according to me. Great work could not find any issues.
Spacing is great as well your naming conventions are meaningful and easy.
File structure is also good. 
Great work.

File: src/context/historyContext.js (hunk 1 above)
Comment (rutu14) [reply src/context/historyContext.js:21]: Sure

**Pull request merged**